Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -55,11 +56,23 @@
 
 namespace WebCore {
 
+static DestinationColorSpace deduceColorSpaceFromCanvas(const SkCanvas& canvas)
+{
+    auto colorSpace = canvas.imageInfo().refColorSpace();
+    if (!colorSpace) {
+        if (auto surface = canvas.getSurface(); surface != nullptr)
+            colorSpace = surface->imageInfo().refColorSpace();
+    }
```

Comment:
In this case I'm not sure, but I suspect SkCanvas::imageInfo() returns the same value as its surface.

Reply:
In that particular case I've actually encountered the case where `canvas.imageInfo().refColorSpace()` is null and yet `surface->imageInfo().refColorSpace()` is not

Reply:
I've added extra comment here:

> Skia sometimes fail to propagate color space info correctly, so we need to fetch few places.

It turns out that on some occasions skia won't propagate the color space information to `SkCanvas`. Actually this was the problem that was leading to lack of conversion when performing `drawImageRect()` on canvas. I've fixed that by specifying color space explicitly in https://github.com/WebKit/WebKit/pull/27079/files#diff-e07ba6513cad2cefec7a94fd9bdf37e71ea228e3f3570c9be1d4aed9dbadd877R82 so that when we're creating surface using e.g. `SkSurfaces::RenderTarget()` the info with non-null color space is used.

I've update other `SkImageInfo`'s constructing calls, but on some occasions `SkCanvas` still has image info out-of-sync with the surface. Looks like some kind of skia bug, so the above change will still be necessary.

Reply:
hmm, so even if explicitly adding SkColorSpace::MakeSRGB() doesn't cover all the cases, I think it's better to simply handle the null colorspace here and default to sRGB, no? We might add new `SkImageInfo::Make` calls and forget about the color space parameter.

Reply:
That would still break `SkCanvas::drawImageRect()`. As suggested offline, we will discuss this matter with skia devs directly.

Reply:
To sum up, the quote from skia dev:

> You should definitely be passing sRGB explicitly (not nullptr), when creating the surface. For sources (eg, images), nullptr is interpreted as sRGB. But for destinations (surfaces), nullptr actually means: "do no color management". This is an ugly legacy option

As for the above code for fetching color space info, again quote from skia dev:

> The discrepancy between the canvas and surface (assuming those are the same underlying surface) is definitely unexpected

So, I'll try to align our `SkImageInfo::Make` calls with what's been mentioned above.